airbyte-workers/src/main/java/io/airbyte/workers/protocols/MessageTracker.java

```diff
@@ -37,6 +37,6 @@
 
   long getBytesCount();
 
-  Optional<JsonNode> getOutputState();
+  Optional<State> getOutputState();
```

Comment:
using the State struct, which is just a wrapper around whatever state the connector returns allows us to differentiate between the case when state is _not_ successfully returned and when state _is_ successfully returned but it is null / empty.